crates/scroll/alloy/evm/src/block/galileo_v2.rs

```
@@ -0,0 +1,222 @@
+
+/// Galileo slot is set to 1 (true) after the `GalileoV2` block fork.
+const IS_GALILEO: U256 = U256::from_limbs([1, 0, 0, 0]);
+
+/// Storage update of L1 gas price oracle at `GalileoV2` transition.
+const GALILEO_V2_L1_GAS_PRICE_ORACLE_STORAGE: [(U256, U256); 1] =
+    [(GPO_IS_GALILEO_SLOT, IS_GALILEO)];
+
+/// Applies the Scroll `GalileoV2` hard fork to the state:
+///    - Updates the L1 oracle contract bytecode.
+///    - Sets the `isGalileo` slot to 1 (true).
+pub(super) fn apply_galileo_v2_hard_fork<DB: Database>(
+    state: &mut State<DB>,
+) -> Result<(), DB::Error> {
+    // No-op if already applied.
+    // Note: This requires a storage read for every block after `GalileoV2`, and it means this
+    // read needs to be included in the execution witness. Unfortunately, there is no
+    // other reliable way to apply the change only at the transition block, since
+    // `ScrollBlockExecutor` does not have access to the parent timestamp.
+    if state.storage(L1_GAS_PRICE_ORACLE_ADDRESS, GPO_IS_GALILEO_SLOT)? == IS_GALILEO {
+        return Ok(())
+    }
+
+    let oracle = state.load_cache_account(L1_GAS_PRICE_ORACLE_ADDRESS)?;
+
+    // compute the code hash
+    let bytecode = Bytecode::new_raw(GALILEO_V2_L1_GAS_PRICE_ORACLE_BYTECODE);
+    let code_hash = bytecode.hash_slow();
+
+    // get the old oracle account info
+    let old_oracle_info = oracle.account_info().unwrap_or_default();
```

This `unwrap_or_default` should work well in practice (we do the same in Feynman), but it looks wrong, should we return error instead of default?

---

I agree, I think returning default is incorrect. We should return an error that results in node shutdown or panic using `expect()` as this suggests the state of the chain is invalid and we can not recover from this.